python/cugraph/tests/test_random_walks.py

```diff
@@ -0,0 +1,154 @@
+# Copyright (c) 2020-2021, NVIDIA CORPORATION.:
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import gc
+
+import pytest
+
+from cugraph.tests import utils
+import cugraph
+import random
+
+
+# =============================================================================
+# Parameters
+# =============================================================================
+DIRECTED_GRAPH_OPTIONS = [False, True]
+WEIGHTED_GRAPH_OPTIONS = [False, True]
+DATASETS = [pytest.param(d) for d in utils.DATASETS]
+DATASETS_SMALL = [pytest.param(d) for d in utils.DATASETS_SMALL]
+
+
+def calc_random_walks(
+    graph_file,
+    directed=False,
+    max_depth=None
+):
+    """
+    compute random walks for each nodes in 'start_vertices'
+
+    parameters
+    ----------
+    G : cuGraph.Graph or networkx.Graph
+        The graph can be either directed (DiGraph) or undirected (Graph).
+        Weights in the graph are ignored.
+        Use weight parameter if weights need to be considered
+        (currently not supported)
+
+    start_vertices : int or list or cudf.Series
+        A single node or a list or a cudf.Series of nodes from which to run
+        the random walks
+
+    max_depth : int
+        The maximum depth of the random walks
+
+
+    Returns
+    -------
+    random_walks_edge_lists : cudf.DataFrame
+        GPU data frame containing all random walks sources identifiers,
+        destination identifiers, edge weights
+
+    seeds_offsets: cudf.Series
+        Series containing the starting offset in the returned edge list
+        for each vertex in start_vertices.
+    """
+    G = utils.generate_cugraph_graph_from_file(
+        graph_file, directed=directed, edgevals=True)
+    assert G is not None
+
+    k = random.randint(1, 10)
+    start_vertices = random.sample(range(G.number_of_vertices()), k)
+    df, offsets = cugraph.random_walks(G, start_vertices, max_depth)
+
+    return df, offsets, start_vertices
+
+
+def check_random_walks(df, offsets, seeds, df_G=None):
+    invalid_edge = 0
+    invalid_seeds = 0
+    invalid_weight = 0
+    offsets_idx = 0
+    for i in range(len(df.index)):
+        src, dst, weight = df.iloc[i].to_array()
+        if i == offsets[offsets_idx]:
+            if df['src'].iloc[i] != seeds[offsets_idx]:
+                invalid_seeds += 1
+                print(
```

Comment:
This can be a FIXME, but tests probably shouldn't rely on print statements to show failures (ie. they should use specific assertions).  I see that this allows you to check every path instead of stopping on the first failure, so we may need to rethink how the test works if we want both assertions instead of prints, and having it not stop on the first failure.

Reply:
I borrowed the idea from test_BFS. I can fix and Fail the test when the first error appears. I liked this approach because the test does not crash at the first failure and I get to see the other mismatches to see a pattern when debugging.

Reply:
I agree that can be nice when you want to see everything. A FIXME will let us revisit this later in a way that can give us both assertions and the ability to see other mismatches if you'd rather do that.